code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -534,70 +534,190 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
+	 * If all this counting fucks up, we revert first to Newton's approximation, then the old simple formula.
+	 */
+
+	// Our thermal energy and moles
+	var/w2 = thermal_energy()
+	var/n2 = total_moles()
+	var/c2 = heat_capacity()
+	
+	// Target thermal energy and moles
+	var/w1 = output_air.thermal_energy()
+	var/n1 = output_air.total_moles()
+	var/c1 = output_air.heat_capacity()
+
+	/// The PV/R part in our equation.
+	var/pvr = pv / R_IDEAL_GAS_EQUATION
+	
+	/// x^2 in the quadratic
+	var/a_value = w2/n2
+	/// x^1 in the quadratic
+	var/b_value = ((n1*w2)/n2) + w1 - (pvr*c2/n2)
+	/// x^0 in the quadratic
+	var/c_value = (-1*pvr*c1) + n1 * w1
+	
+	. = gas_pressure_quadratic(a_value, b_value, c_value, lower_limit, upper_limit)
+	if(.)
+		return
+	. = gas_pressure_approximate(a_value, b_value, c_value, lower_limit, upper_limit)
+	if(.)
+		return
+	// Inaccurate and will probably explode but whatever.
+	return (pressure_delta*output_air.volume)/(temperature * R_IDEAL_GAS_EQUATION)
+
+/// Actually tries to solve the quadratic equation.
+/// Do mind that the numbers can get very big and might hit BYOND's single point float limit.
+/datum/gas_mixture/proc/gas_pressure_quadratic(a, b, c, lower_limit, upper_limit)
+	if(IS_INF_OR_NAN(a) || IS_INF_OR_NAN(b) || IS_INF_OR_NAN(c))
+		stack_trace("Failed to solve pressure quadratic equation, inf-nan check failed. A: [a]. B: [b]. C:[c]. Expected lower limit: [lower_limit]. Expected upper limit: [upper_limit].")
 		return FALSE
+	var/solution = max(SolveQuadratic(a, b, c)) 
+	if((solution > lower_limit) && (solution < upper_limit)) //SolveQuadratic can return nulls so be careful here
+		return solution
+	stack_trace("Failed to solve pressure quadratic equation. A: [a]. B: [b]. C:[c]. Current value = [solution].  Expected lower limit: [lower_limit]. Expected upper limit: [upper_limit].")
+	return FALSE
 
-	//Calculate necessary moles to transfer using PV=nRT
-	if((total_moles() > 0) && (temperature>0))
-		var/pressure_delta = target_pressure - output_starting_pressure
-		var/transfer_moles = (pressure_delta*output_air.volume)/(temperature * R_IDEAL_GAS_EQUATION)
-
-		//Actually transfer the gas
-		if(specific_gas)
-			var/datum/gas_mixture/removed = remove_specific(specific_gas, transfer_moles)
-			output_air.merge(removed)
-			return TRUE
-		var/datum/gas_mixture/removed = remove(transfer_moles)
-		output_air.merge(removed)
-		return TRUE
+/// Approximation of the quadratic equation using Newton-Raphson's Method.
+/// We use the slope of an approximate value to get closer to the root of a given equation.
+/datum/gas_mixture/proc/gas_pressure_approximate(a, b, c, lower_limit, upper_limit)
+	if(IS_INF_OR_NAN(a) || IS_INF_OR_NAN(b) || IS_INF_OR_NAN(c))
+		stack_trace("Newton's Approximation for pressure failed, inf-nan check failed. A: [a]. B: [b]. C:[c]. Expected lower limit: [lower_limit]. Expected upper limit: [upper_limit]. ")
+		return FALSE
```

Comment:
I hadn't considered this properly. If you'd like you can keep the nesting

Reply:
👍